### ISSUE TITLE
cargo build fails to parse gitconfig

### ISSUE THREAD

**Issue by portiatrivisonno** [rust-lang/cargo#9853]
**Problem**
Cargo fails to parse a valid gitconfig file and becomes unusable (see also #9854)

MWE1
```ini
[alias]
	myalias1 = log --pretty='case foo in\
*) sign=-S ";;" \
*) sign=%h ";;" \
esac \
'
```

MWE2:
```ini
[alias]
	myalias2 = cmd \
";;" \
";;" \
bar
```

The error is indicated to be in line 5 and line 4, respectively, with
```
failed to parse config file: invalid configuration key (in ~/.config/git/config:5); class=Config (7)
```

Git has no problem using these.  I condensed the MWE from an alias of a `filter-branch` which takes as argument shell code, similar to MWE1.

**Notes**

Output of `cargo version`: 1.53.0

**Comment by ehuss** [comment]
Thanks, I have filed the upstream issue here: https://github.com/libgit2/libgit2/issues/6019

**Comment by basile-henry** [comment]
The parsing error is now fixed upstream: https://github.com/libgit2/libgit2/pull/6043

Just need to wait for the next release 😄

**Comment by portiatrivisonno** [comment]
The libgit2 fix (https://github.com/libgit2/libgit2/pull/6043) is included in libgit2 >=1.3.0 (released Sep 27, 2021), so I think this issue can be closed.  Thanks for taking care of this and forwarding it to upstream!

**Comment by ehuss** [comment]
Yep, thanks!
